hibernate-reactive-core/src/test/java/org/hibernate/reactive/schema/NumericValidationTest.java

```diff
@@ -0,0 +1,185 @@
+
+public abstract class NumericValidationTest extends BaseReactiveTest {
+
+	public static class IndividuallyNumericValidationTestBase extends NumericValidationTest {
+
+		@Override
+		protected Configuration constructConfiguration(String hbm2DdlOption) {
+			final Configuration configuration = super.constructConfiguration( hbm2DdlOption );
+			configuration.setProperty( Settings.HBM2DDL_JDBC_METADATA_EXTRACTOR_STRATEGY, INDIVIDUALLY.toString() );
+			return configuration;
+		}
+	}
+
+	public static class GroupedNumericValidationTestBase extends NumericValidationTest {
+
+		@Override
+		protected Configuration constructConfiguration(String hbm2DdlOption) {
+			final Configuration configuration = super.constructConfiguration( hbm2DdlOption );
+			configuration.setProperty( Settings.HBM2DDL_JDBC_METADATA_EXTRACTOR_STRATEGY, GROUPED.toString() );
+			return configuration;
+		}
+	}
+
+	protected Configuration constructConfiguration(String hbm2DdlOption) {
+		Configuration configuration = constructConfiguration();
+		configuration.setProperty( Settings.HBM2DDL_AUTO, hbm2DdlOption );
+		configuration.setProperty( Settings.DEFAULT_CATALOG, "hreact" );
+		return configuration;
+	}
+
+	@Before
+	@Override
+	public void before(TestContext context) {
+
+		Configuration configuration = constructConfiguration( "create" );
+		configuration.addAnnotatedClass( TestEntity.class );
+
+		test( context, setupSessionFactory( configuration )
+				.thenCompose( v -> factoryManager.stop() ) );
+	}
+
+	@After
+	@Override
+	public void after(TestContext context) {
+
+		final Configuration configuration = constructConfiguration( "drop" );
+		configuration.addAnnotatedClass( TestEntity.class );
+
+		test( context, factoryManager.stop()
+				.thenCompose( v -> setupSessionFactory( configuration ) )
+				.thenCompose( v -> factoryManager.stop() ) );
+	}
+
+	@Test
+	public void testValidationSucceed(TestContext context) {
+		Configuration configuration = constructConfiguration( "validate" );
+		configuration.addAnnotatedClass( TestEntity.class );
+
+		test(
+				context,
+				setupSessionFactory( configuration )
+						.thenCompose( v -> getSessionFactory()
+								.withTransaction(
+										(session, t) -> session.createQuery( "FROM TestEntity", TestEntity.class )
+												.getResultList() ) )
+						.thenAccept( results -> context.assertTrue( results.isEmpty() ) )
+		);
+	}
+
+	@Test
+	public void testDataTypePostgreSql(TestContext context) {
+		final TestEntity testEntity = new TestEntity();
+		testEntity.id = 9;
+		testEntity.number = new BigDecimal( 3.14159 );
+
+		String columnTypeQuery =
+				"select data_type from information_schema.columns " +
+						"where table_name = 'testentity' and column_name = 'numbervalue';";
+
+		final Configuration configuration = constructConfiguration( "update" );
+		configuration.addAnnotatedClass( TestEntity.class );
+
+		if ( DatabaseConfiguration.dbType() != DatabaseConfiguration.DBType.POSTGRESQL ) {
+			return;
+		}
```

Comment:
If you do things this way, this test will result in a success even if, in reality, you want to skip it.
That's way I've created the rule. It will also call the before/after methods.

Reply:
Right.   Some of these tests are not DB specific in their purpose, but DB specific in **query** implementation.  So options are still to find a cleaner (efficient) way to make that happen in one test class, or break them apart into multiple classes. I'm good with either.

Reply:
Break it in separate classes, please. Maybe inner classes extending an abstract super class.

The rule I've created is not that powerful and it only works for classes.  if the only difference is the query, having an abstract super class with a mehod `getQuery` (and maybe `getResult`) will avoid a lot of duplicated work,

I think we could now switch to JUnit 5 and it will make the management of different test sper database easier. But that's a separate issue for when we have time.

Reply:
sure.. I'll  work toward abstraction and getQuery() & getResult().

@DavideD I added a `schema` package since there may be more tests.  Any issue with doing that at this time?

Reply:
> @DavideD I added a `schema` package since there may be more tests. Any issue with doing that at this time?

No, that's fine

Reply:
Just keep in mind that we already have several test classes in the project and you should move them in the sama package if you want to create a new one